File: mc_raylib/src/block.rs
use raylib::prelude::*;

use crate::block_type::BlockType;
use crate::dir::Dir;

#[derive(Clone, Copy)]
pub struct Block {
    pub block_type: BlockType,
    pub get_color: fn(&Dir) -> Color,
    pub transparent: bool,
}
impl Block {
    pub fn new(block_type: BlockType) -> Self {
        Self {
            block_type,
            get_color: block_type.get_color_fn(),
            transparent: block_type.get_transparent(),
        }
    }
}
impl Default for Block {
    fn default() -> Self {
        let default_block_type = BlockType::Air;
        Self {
            block_type: default_block_type,
            get_color: default_block_type.get_color_fn(),
            transparent: default_block_type.get_transparent(),
        }
    }
}

File: mc_raylib_cpp/block.cpp
#include "raylib.h"

#include "include/blockType.h"

#include "include/block.h"


Block::Block(const BlockType& blockType) {
	this->blockType = blockType;
	this->transparent = getTransparent(blockType);
	this->solid = getSolid(blockType);
	this->getTexcoords = getTexcoordsFn(blockType);
}